src/marqo/tensor_search/utils.py

```diff
@@ -351,3 +352,11 @@ def generate_batches(seq: Sequence, batch_size: int):
 
     for i in range(0, len(seq), batch_size):
         yield seq[i:i + batch_size]
+
+def get_best_available_device() -> str:
+    device = read_env_vars_and_defaults(EnvVars.MARQO_BEST_AVAILABLE_DEVICE)
+    if (device is None) or not (device == "cpu" or device.startswith("cuda")):
```

Comment:
I don't want to call he API validation here. So i just do a simple check here @pandu-k

Reply:
you can use this for validation: https://github.com/marqo-ai/marqo/blob/mainline/src/marqo/tensor_search/utils.py#L117

To keep our device validation code in one place